Boot 2 Logo

Jakewd opened the pull request.

Jakewd committed:
Update readme.md

added the boot logo instructions and suggested software needed to edit the image

Jakewd committed:
Added a boot logo option

Must use a binary (.bin) image renamed to bootlogo.tns. This is NOT my creation I have just simply Proposed it as a change because it may be useful to some people. My credits are to ZephRay and her article on cncalc (http://www.cncalc.org/forum.php?mod=viewthread&tid=9774). There it illustrates how to create the binary file from a regular image.

Vogtinator commented on src/nlaunch.c:
```diff
@@ -207,11 +207,20 @@ static __attribute__((always_inline, noreturn)) void launch_OS(void) {
+//! Display BootImage.tns
+static __attribute__((always_inline)) void display_bootlogo(void) {
+	FILE *logo;
+	logo = fopen((char *)osbootlogo, "rb");
```
This will just crash if the file does not exist.

Jakewd committed:
Update nlaunch.c

would this prevent a crash from occurring? I saw that you did that same if statement when you created the function to update nlaunchy if needed.

Jakewd reviewed (commented):
Would this fix the issue of crashing

Axenntio commented on src/nlaunch.c:
```diff
@@ -207,11 +207,21 @@ static __attribute__((always_inline, noreturn)) void launch_OS(void) {
+//! Display BootImage.tns
+static __attribute__((always_inline)) void display_bootlogo(void) {
+	FILE *logo;
+	if ((logo = fopen((char *)osbootlogo, "rb"))); {
+	    fread(*(volatile unsigned*)0xC0000010,buffer_size,1,logo);
```
Why is there a `;` at the end of the condition?
`buffer_size` is not define
